apiclient.go

```diff
@@ -1,25 +1,23 @@
-package client
+package warrant
 
 import (
 	"bytes"
 	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
-
-	"github.com/warrant-dev/warrant-go/v4/config"
 )
 
 const (
-	ClientVersion string = "3.1.0"
+	ClientVersion string = "4.0.0"
 )
 
 type WarrantClient struct {
 	HttpClient *http.Client
-	Config     config.ClientConfig
+	Config     ClientConfig
 }
 
-func (client WarrantClient) MakeRequest(method string, path string, payload interface{}) (*http.Response, error) {
+func (client WarrantClient) MakeRequest(method string, path string, payload interface{}, options *RequestOptions) (*http.Response, error) {
```

Comment:
Can you simplify this function? Instead, can we only set the request body if `payload != nil` and then set the headers after that?

Reply:
```
func (client WarrantClient) MakeRequest(method string, path string, payload interface{}, options *RequestOptions) (*http.Response, error) {
        var requestBody
        if payload != nil {
	        postBody, err := json.Marshal(payload)
	        if err != nil {
		        return nil, WrapError("Invalid request payload", err)
	        }
	        requestBody := bytes.NewBuffer(postBody)
        }
	req, err := http.NewRequest(method, url, requestBody)
	if err != nil {
		return nil, WrapError("Unable to create request", err)
	}
        url := client.Config.ApiEndpoint + path
	if client.Config.ApiKey != "" {
		req.Header.Add("Authorization", fmt.Sprintf("ApiKey %s", client.Config.ApiKey))
	}
	if options.WarrantToken != "" {
		req.Header.Add("Warrant-Token", options.WarrantToken)
	}
	req.Header.Add("User-Agent", fmt.Sprintf("warrant-go/%s", ClientVersion))
	resp, err := client.HttpClient.Do(req)
	if err != nil {
		return nil, WrapError("Error making request", err)
	}
	respStatus := resp.StatusCode
	if respStatus < 200 || respStatus >= 400 {
		msg, _ := io.ReadAll(resp.Body)
		return nil, Error{
			Message: fmt.Sprintf("HTTP %d %s", respStatus, string(msg)),
		}
	}
	return resp, nil
}
```

Reply:
Some context for my change: I have to have separate paths for the request instance because `NewRequest` does this:
```
switch v := body.(type) {
		case *bytes.Buffer:
``` 

so when we pass in a nil pointer of type `*bytes.Buffer`, we run into a segmentation fault. That probably also explains why I initially had two separate paths but I'm reusing the rest of the request configuration logic.